sched/scheduler/job_state.go

```diff
@@ -61,14 +64,24 @@ func newJobState(job *sched.Job, saga *saga.Saga) *jobState {
 	// are considered not done and will be rescheduled.
 	for _, taskId := range saga.GetState().GetTaskIds() {
 		if saga.GetState().IsTaskCompleted(taskId) {
-			j.Tasks[taskId].Status = sched.Completed
+			j.getTask(taskId).Status = sched.Completed
 			j.TasksCompleted++
 		}
 	}
 
 	return j
 }
 
+// Helper, assumes that taskId is present given a consistent jobState.
```

Comment:
I'm a little worried about performance.... we could maintain the map and make sure the tasks are run in their original order if we generate task id's from the job id: <job_id>_1, <job_id>_2, ... , <job_id>_<numTasks -1> 
(just a thought)